varnish/pipedream-redirects.changelog.com.vcl

```diff
@@ -0,0 +1,119 @@
+sub vcl_recv {
+
+    # redirects
+    # /uploads/podcast/news-2022-06-27/the-changelog-news-2022-06-27.mp3 /uploads/news/1/changelog-news-1.mp3
+    # /uploads/podcast/news-2022-07-04/the-changelog-news-2022-07-04.mp3 /uploads/news/2/changelog-news-2.mp3
+    # /uploads/podcast/news-2022-07-11/the-changelog-news-2022-07-11.mp3 /uploads/news/3/changelog-news-3.mp3
+
+    # testing
+    # httpstat "http://localhost:9000/uploads/podcast/news-2022-06-27/the-changelog-news-2022-06-27.mp3"
+    # httpstat "http://localhost:9000/uploads/podcast/news-2022-06-27/the-changelog-news-2022-06-27.mp3?this=is&a=query&string"
+
+    # comparison
+    # httpstat "http://www.changelog.com/uploads/podcast/news-2022-06-27/the-changelog-news-2022-06-27.mp3"
+    # httpstat "http://www.changelog.com/uploads/podcast/news-2022-06-27/the-changelog-news-2022-06-27.mp3?this=is&a=query&string"
+
+    # Remove X-Redirect header from client requests to prevent header injection
+    unset req.http.X-Redirect;
+
+    if (req.url ~ "^/uploads/podcast/news-2022-06-27/the-changelog-news-2022-06-27.mp3($|\?)") {
```

Comment:
So we don't lose the conversation we had: Would it be more performant to wrap this chain of reg-ex's with a simple prefix check, so that they all get skipped if the prefix makes it impossible.